### PR TITLE
RN-802 Default parameters on preview

### PR DESCRIPTION
**Pull request by biaoli0**
### Issue #:
RN-802
 
### Changes:
1. Add a couple of Fields for builtInParams, see: packages/admin-panel/src/dataTables/components/PreviewFilters/filters/FilterTypeOptions.js
2. Rename variables and functions from *Parameters (e.g. useParameters) to *Params to be consistent with data-table-server
3. Refactor Autocomplete to ReduxAutocomplete

### CODE REVIEW

File: packages/admin-panel/src/autocomplete/Autocomplete.js [hunk 1]
```diff
@@ -1,198 +1,133 @@
...
+        renderTags: (values, getTagProps) =>
+          values.map((option, index) => (
+            <Chip
+              color="primary"
+              label={optionLabelKey ? option[optionLabelKey] : option}
+              {...getTagProps({ index })}
+            />
+          )),
+      }
+    : {};
```
Comment (biaoli0) [inline packages/admin-panel/src/autocomplete/Autocomplete.js:76]: These parts can be reused for multiple selections.

File: packages/admin-panel/src/autocomplete/Autocomplete.js [hunk 2]
```diff
@@ -1,198 +1,133 @@
...
-    return Array.isArray(placeholder) ? placeholder.join(', ') : placeholder;
-  }
-  return 'Start typing to search';
-};
-
-const filter = createFilterOptions();
-
-const AutocompleteComponent = React.memo(
-  ({
+export const Autocomplete = props => {
```
Comment (biaoli0) [inline packages/admin-panel/src/autocomplete/Autocomplete.js:19]: This components will accept options from props. This brings some flexibilities to fetch options either from Redux dispatch or use query.

File: packages/admin-panel/src/autocomplete/Autocomplete.js [hunk 3]
```diff
@@ -1,198 +1,133 @@
...
-import throttle from 'lodash.throttle';
-import { connect } from 'react-redux';
-import MuiChip from '@material-ui/core/Chip';
 import { createFilterOptions } from '@material-ui/lab/Autocomplete';
 import styled from 'styled-components';
-import { Autocomplete as AutocompleteBase } from '@tupaia/ui-components';
-import { getAutocompleteState } from './selectors';
-import { changeSelection, changeSearchTerm, clearState } from './actions';
+import MuiChip from '@material-ui/core/Chip';
+import { Autocomplete as UIAutocomplete } from '@tupaia/ui-components';
```
Comment (chris-pollard) [inline packages/admin-panel/src/autocomplete/Autocomplete.js:11]: I wonder if by convention, the name should be UiAutocomplete. Having the first word in caps doesn't seem to fall in line with the way we name react components

**Review by chris-pollard — APPROVED** [review]
Nice refactoring work!

File: packages/admin-panel/src/autocomplete/Autocomplete.js (hunk 3 above)
Comment (biaoli0) [reply packages/admin-panel/src/autocomplete/Autocomplete.js:11]: We will do that when it comes to abbreviation, e.g. PDFExportPage

**Pull request merged**